### PR TITLE
Replace hacky_postprocess_around_closure_limitations.py

### PR DESCRIPTION
**Pull request by juj**
Replace regex-based tools/hacky_postprocess_around_closure_limitations.py with an acorn AST-based unsafe optimizer.

### CODE REVIEW

File: tools/unsafe_optimizations.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
+const acorn = require('acorn');
```
Comment (sbc100) [inline tools/unsafe_optimizations.js:1]: How about something like this at the top"

```
#!/usr/bin/env node                                                                                
 
/** Top level comment about what this scripte is for and how / when it run */
                                                                                   
'use strict'; 
```

File: tools/unsafe_optimizations.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    console.error('----NODE----');
+    console.error(node);
+    console.error('----ENDNODE----');
+    assert(false);
+  }
+  return children;
+}
+
+function dump(nodeArray) {
+  for(let node of nodeArray) {
```
Comment (sbc100) [inline tools/unsafe_optimizations.js:64]: space after `for`, here and elsewhere in this file

File: tools/unsafe_optimizations.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      assert(!Array.isArray(child));
+      children.push(child);
+    }
+  }
+  function addChildArray(childArray) {
+    assert(Array.isArray(childArray));
+    children = children.concat(childArray);
+  }
+
+  if (['BlockStatement', 'Program'].indexOf(node.type) != -1) { addChildArray(node.body); }
```
Comment (sbc100) [inline tools/unsafe_optimizations.js:26]: Con you use `contains()` here to make all these a little more concise:

``` if (['BlockStatement', 'Program'].contains(node.type) {```

File: tests/code_size/random_printf_wasm2js.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
-  "a.html": 17794,
-  "a.html.gz": 7593,
-  "total": 17794,
-  "total_gz": 7593
+  "a.html": 17804,
```
Comment (juj) [inline tests/code_size/random_printf_wasm2js.json:2]: This is the only code size that increases, but when I locally do a build of this test case on current `main` vs this branch, I get exactly identical results to the byte, so I wonder if the old numbers weren't up to date..

Comment (sbc100) [reply tests/code_size/random_printf_wasm2js.json]: Indeed, there is some tolerance in the test code so the old number are rarely 100% correct.

What I normally do when I really want to see the impact of a change is run `./tests/runner other.*code_size* other.*metadce* --rebase` on `origin/main` and submit that change before doing the same on my work branch.

You can see several such changes in the git history where we just bring the current numbers up-to-date. e.g: de4cb659f9653777d2bdb33c44aab73f65749c74

Comment (juj) [reply tests/code_size/random_printf_wasm2js.json]: Looks like in the latest the code size here also decreases, adjusted to do optimization a bit wider. Also restored the code size test on Windows and macOS, they do seem to currently agree with each other.

Comment (sbc100) [reply tests/code_size/random_printf_wasm2js.json]: Thats great news!   I was worried that debugging that would be pain

**Review by kripken — APPROVED** [review]
Code looks reasonable to me. Ideally maybe this would be a set of new passes in the existing acorn-optimizer.js file, but as this is just for MINIMAL_RUNTIME=2 then I am less worried, and it's a definitely improvement over the old code obviously.

It looks like we have very minimal testing of this mode, btw? Might be worth doing more manual testing for this rewrite if you haven't already.

**Pull request merged**